### ISSUE TITLE
Deprecate `QubitMapper.allows_two_qubit_reduction`

### ISSUE THREAD

**Issue by mrossinek** [Qiskit/qiskit-nature#1039]
### What should we add?

Once #1018 is merged, the attribute `QubitMapper.allows_two_qubit_reduction` is no longer needed and should be deprecated.
When doing this, we need to carefully double check any implications this might have for the public API.
